snuba/migrations/runner.py

```diff
@@ -22,44 +25,70 @@ class Status(Enum):
     COMPLETED = "completed"
 
 
+class MigrationKey(NamedTuple):
+    group: MigrationGroup
+    migration_id: str
+
+    def __str__(self) -> str:
+        return f"{self.group.value} {self.migration_id}"
+
+
 class Runner:
-    def run_migration(self, group: MigrationGroup, migration_id: str) -> None:
+    def __init__(self) -> None:
+        self.__connection = get_cluster(StorageSetKey.MIGRATIONS).get_query_connection(
+            ClickhouseClientSettings.MIGRATE
+        )
+
+    def run_migration(self, migration_key: MigrationKey) -> None:
         """
-        Run a single migration given its ID and marks the migration as complete.
+        Run a single migration given its migration key and marks the migration as complete.
         """
+        group = migration_key.group
+        migration_id = migration_key.migration_id
+
         assert all(
             cluster.is_single_node() for cluster in CLUSTERS
         ), "Cannot run migrations for multi node clusters"
 
-        logger.info(f"Running migration: {group} {migration_id}")
+        logger.info(f"Running migration: {migration_key}")
 
         migration = get_group_loader(group).load_migration(migration_id)
 
+        self._update_migration_status(migration_key, Status.IN_PROGRESS)
```

Comment:
At which point do you expect that we will load the current status of the migration to ensure we are in a state compatible with starting it from scratch ? Would that be here and we would stop if the migration is already in the database flagged as "in progress" ?

Reply:
I think we'll probably need to do that upfront before this "run_migration" method gets called and refuse to run if any migration is found to be in progress. The user will probably need to do something else to fix/complete/reverse the migration if it's in progress.